pyppeteer/page.py

```diff
@@ -454,20 +479,13 @@ def _onCertificateError(self, event: Any) -> None:
 
         :arg str expression: XPath string to be evaluated.
         """
-        frame = self.mainFrame
-        if not frame:
-            raise PageError('no main frame.')
-        return await frame.xpath(expression)
+        return await self.mainFrame.xpath(expression)
 
-    #: alias to :meth:`querySelector`
+    # Shortcut aliases
     J = querySelector
-    #: alias to :meth:`querySelectorEval`
     Jeval = querySelectorEval
-    #: alias to :meth:`querySelectorAll`
     JJ = querySelectorAll
-    #: alias to :meth:`querySelectorAllEval`
     JJeval = querySelectorAllEval
-    #: alias to :meth:`xpath`
     Jx = xpath
```

Comment:
Makes more senses to assign these at class instantiation

Reply:
I've been thinking about that - assigning in init doesn't expose the API if the object is not initiated which might fail automcomplete/inspection on some editors as well as making it harder to monkeypatch.

Reply:
Fair enough. Another thing is that docs builds won't handle it as well.